pennylane/devices/default_qubit.py

```diff
@@ -379,6 +380,21 @@ def name(self):
         """The name of the device."""
         return "default.qubit"
 
+    @property
+    def prng_key(self):
+        """Get a new key with ``jax.random.split``."""
+        self._prng_key, key = jax_random_split(self._prng_key)
+        return key
+
+    def get_prng_key(self, num: int = 1):
+        """Get ``num`` new keys with ``jax.random.split``."""
+        self._prng_key, *keys = jax_random_split(self._prng_key, num=num + 1)
```

Comment:
Why the num+1?

Reply:
This is because whenever requesting a key, we must generate at least an extra key which is used to spawn another tree of keys; the first key is used to overwrite `_prng_key` for that purpose. Otherwise there is a risk of reusing any of the keys. This is done under the hood to prevent users from forgetting to update the "seed".

Reply:
Nice 👍